src/net.h

```diff
@@ -865,6 +865,88 @@ class CConnman
     /** Return true if we should disconnect the peer for failing an inactivity check. */
     bool ShouldRunInactivityChecks(const CNode& node, std::chrono::seconds now) const;
 
+    /**
+     * Placeholder for total network traffic. Split by direction, network, connection
+     * type and message type (byte and message counts).
+     */
+    class NetStats
+    {
+    public:
+        struct MsgStat {
+            std::atomic_uint64_t byte_count; //!< Number of bytes transferred.
+            std::atomic_uint64_t msg_count;  //!< Number of messages transferred.
+
+            MsgStat() : byte_count{0}, msg_count{0} {}
+
+            MsgStat(const MsgStat& x) : byte_count{x.byte_count.load()}, msg_count{x.msg_count.load()} {}
+
+            MsgStat(std::atomic_uint64_t x, std::atomic_uint64_t y) : byte_count{x.load()}, msg_count{y.load()} {}
+        };
+
+        enum class Direction { SENT,
+                               RECV };
+
+        /// Number of elements in `Direction`.
+        static constexpr size_t NUM_DIRECTIONS{2};
+
+        using MultiDimensionalStats = std::array<std::array<std::array<std::array<MsgStat,
+                                                                                  // add 1 for the other message type
```

Comment:
Would be easier if the `*other*` message type was added to the `allNetMessageTypes` array (and corresponding `NUM_NET_MESSAGE_TYPES`) in `protocol.cpp`?